x-pack/plugins/dataset_quality/public/hooks/use_dataset_quality_flyout.tsx

```diff
@@ -5,60 +5,38 @@
  * 2.0.
  */
 
-import { useMemo } from 'react';
-import { useFetcher } from '@kbn/observability-shared-plugin/public';
-import { DataStreamStatServiceResponse } from '../../common/data_streams_stats';
-import { DataStreamNameParts, dataStreamPartsToIndexName } from '../../common/utils';
+import { useSelector } from '@xstate/react';
 import { useDatasetQualityContext } from '../components/dataset_quality/context';
 import { useKibanaContextForPlugin } from '../utils';
 
-export const useDatasetQualityFlyout = ({ type, dataset, namespace }: DataStreamNameParts) => {
+export const useDatasetQualityFlyout = () => {
   const {
     services: { fieldFormats },
   } = useKibanaContextForPlugin();
-  const { dataStreamsStatsServiceClient: client } = useDatasetQualityContext();
-  const { data: dataStreamStatList = [], loading: dataStreamStatLoading } = useFetcher(
-    async () => client.getDataStreamsStats({ datasetQuery: `${dataset}-${namespace}`, type }),
-    [dataset, namespace, type]
-  );
 
-  const { data: dataStreamDetails = {}, loading: dataStreamDetailsLoading } = useFetcher(
-    async () =>
-      client.getDataStreamDetails({
-        dataStream: dataStreamPartsToIndexName({ type, dataset, namespace }),
-      }),
-    [dataset, namespace, type]
+  const { service } = useDatasetQualityContext();
+
+  const isLoaded = useSelector(service, (state) => state.matches('datasets.loaded'));
+
+  const { dataset: dataStreamStat, datasetDetails: dataStreamDetails } = useSelector(
+    service,
+    (state) => state.context.flyout
+  );
+  const dataStreamDetailsLoading = useSelector(service, (state) =>
+    state.matches('datasets.loaded.flyoutOpen.fetching')
   );
 
-  return useMemo(() => {
-    const isDataStreamStatStale = isStaleData({ type, dataset, namespace }, dataStreamStatList[0]);
+  if (isLoaded && !!dataStreamStat && !dataStreamDetails) {
+    service.send({
+      type: 'OPEN_FLYOUT',
+      dataset: dataStreamStat,
+    });
+  }
```

Comment:
**question**: why do we need to open the flyout if not explicitly clicking on the expand button from a table row?

Reply:
You could share your url with another user and they could see exactly the same state you had at the moment of sharing, this includes having the flyout open. We have also discussed potential crosslinking between log document flyout details and this page, so users could navigate directly to this page with the dataset selected (flyout opened)

Reply:
If this depends on the state of the URL, should this action rely on a URL parameter to exist to be triggered on load?

Reply:
The URL is detached from the state, URL is handled by the consumer, in the plugin we only rely on the state. How can we know just from the state whether we should open it or not? What would you suggest?

Reply:
In the ObservabilityDatasetQuality state machine, there is an initialization flow controlled before the machine load:
- Initialize from URL
- Create controller
- Initialized

When the controller is created and in the initialized state, you have control over the controller state machine to trigger a transition depending on the consumer URL state. For instance, you could add a step in the initialized state that looks up in the URL state if there is a parameter that tells to open the flyout and on which dataset, and then triggers the `OPEN_FLYOUT` transition from the controller. By doing so you can control the behaviour of the page controller from the consumer side, and it doesn't constrain each consumer to respect this behaviour if not required.

Reply:
Very nice suggestion ✨
Addressed in [dfae612](https://github.com/elastic/kibana/pull/174906/commits/dfae612e5da1f8ad4bdbc953ada29b6a7007b1a6).